common/sagas/schedule/paramsValidity.ts

```diff
@@ -0,0 +1,154 @@
+  ICurrentScheduleTimestamp,
+  ICurrentTimeBounty,
+  getSchedulingToggle,
+  getScheduleTimestamp,
+  getScheduleTimezone,
+  getScheduleType,
+  getWindowStart,
+  getWindowSize,
+  getTimeBounty,
+  getScheduleGasPrice,
+  getScheduleGasLimit,
+  getScheduleDeposit
+} from 'selectors/schedule/fields';
+import {
+  TypeKeys,
+  SetScheduleParamsValidityAction,
+  setScheduleParamsValidity
+} from 'actions/schedule';
+import { TypeKeys as TransactionTypeKeys } from 'actions/transaction';
+import { getData } from 'selectors/transaction/fields';
+import { getWalletInst } from 'selectors/wallet';
+import {
+  EAC_SCHEDULING_CONFIG,
+  calcEACEndowment,
+  getValidateRequestParamsData,
+  EAC_ADDRESSES,
+  parseSchedulingParametersValidity
+} from 'libs/scheduling';
+import { gasPriceToBase } from 'libs/units';
+import BN from 'bn.js';
+import { bufferToHex } from 'ethereumjs-util';
+import RequestFactory from 'libs/scheduling/contracts/RequestFactory';
+import { windowSizeBlockToMin, calculateWindowStart } from 'selectors/schedule/helpers';
+import { getCurrentTo, getCurrentValue } from 'selectors/transaction/current';
+
+export function* shouldValidateParams(): SagaIterator {
+  while (true) {
+    yield take([
+      TransactionTypeKeys.TO_FIELD_SET,
+      TransactionTypeKeys.DATA_FIELD_SET,
+      TypeKeys.CURRENT_TIME_BOUNTY_SET,
+      TypeKeys.WINDOW_SIZE_FIELD_SET,
+      TypeKeys.WINDOW_START_FIELD_SET,
+      TypeKeys.SCHEDULE_TIMESTAMP_FIELD_SET,
+      TypeKeys.TIME_BOUNTY_FIELD_SET,
+      TypeKeys.SCHEDULE_TYPE_SET,
+      TypeKeys.SCHEDULING_TOGGLE_SET,
+      TypeKeys.SCHEDULE_TIMEZONE_SET
+    ]);
+
+    yield call(delay, 250);
+
+    const isOffline: boolean = yield select(getOffline);
+    const schedulingToggle: ICurrentSchedulingToggle = yield select(getSchedulingToggle);
+    const scheduling = Boolean(schedulingToggle && schedulingToggle.value);
+
+    if (isOffline || !scheduling) {
+      continue;
+    }
+
+    yield call(checkSchedulingParametersValidity);
+  }
+}
+
+function* checkSchedulingParametersValidity() {
+  const currentTo = yield select(getCurrentTo);
+  const currentValue = yield select(getCurrentValue);
+  const callData = yield select(getData);
+  const scheduleType: ICurrentScheduleType = yield select(getScheduleType);
+  const windowStart: ICurrentWindowStart = yield select(getWindowStart);
+  const windowSize: ICurrentWindowSize = yield select(getWindowSize);
+  const timeBounty: ICurrentTimeBounty = yield select(getTimeBounty);
+  const scheduleGasPrice = yield select(getScheduleGasPrice);
+  const scheduleGasLimit = yield select(getScheduleGasLimit);
+  const deposit = yield select(getScheduleDeposit);
+  const node = yield select(getNodeLib);
+  const wallet = yield select(getWalletInst);
+  const scheduleTimestamp: ICurrentScheduleTimestamp = yield select(getScheduleTimestamp);
+  const scheduleTimezone: ICurrentScheduleTimezone = yield select(getScheduleTimezone);
+
+  if (
```

Comment:
it would be nice if large conditionals like these were commented in groups for maintainability, and having this in its own selector

Reply:
Hello, can you help me to understand what does it mean "commented in groups"?